lib/entry-list.ts

```diff
@@ -19,32 +42,39 @@ const runHandler = async (
     }
 
     // await will accept both async and sync functions, no need for the isPromise check
-    return await handler(handlerArg)
+    const result = await handler(handlerArg)
+    return validateResult(result)
   } catch (error) {
     console.error(error)
     return false
   }
 }
 
-export default function createEntryList(channel: Channel): EntryListAPI {
-  let _handler: OnEntryListUpdatedHandler | null = null
+const validateResult = (result: OnEntryListUpdatedHandlerReturn) => {
```

Comment:
It feels strange that the method returns a value which is always identical to the input or throws an error.
I'd make this an assert method that never returns anything but throws if something is wrong.

Now it feels like it's also doing some data cleanup or similar